### PR TITLE
Add sections to the FileSystem module

### PR DESCRIPTION
**Pull request by lydia-duncan**
Brad had pointed out that it would make surfing the FileSystem module easier if
we had a quick reference guide to the different functions and their relative
categories.  To that end, I added four different sections: File/Directory
Manipulations, File/Directory Properties, Locale State Functionality, and
Filerators.  I then added a fifth section for if one just wanted to skip to the
definitions and didn't care about all that organizational stuff.

### CODE REVIEW

**Comment by bradcray** [comment]
I think this is a strong step forward for making the module's contents more approachable.  I'd recommend having someone who's more familiar with these routines and who weighed in heavily on the original proposal to review it as well from a content/categories perspective (and just to get an additional opinion).

Though I've continued to use "Filerator" in my rough drafts and conversation, I think we should move away from this term in the documentation and switch to something like "File System Traversal Iterators".

Within each category, if there were some order for the routines other than "alphabetical" that made sense (say, most to least used or simplest to most complex), I don't know that keeping these lists alphabetical adds that much value.  If we did go away from alphabetical, I think there is a more logical order for filerators, but if it's the only case, then let's stick with alphabetical.

Part of me wishes for a more vertically-oriented way of listing the categories' contents, but doing one category after the next that way would clearly take up too much space.  Do we have any options for making a list of lists without going to a full-blown 2D table?  (which seems like the wrong stylistic idiom).  Sketching using ascii text (and not having it render well), I'm imagining something like:

   `Category    Category   Category    Category`
   `One         Two        Three       Four`
 
   `routine 1   routine 1  routine 1   routine 1`
   `routine 2   routine 2  routine 2   routine 2`

Still, I think that this is spit-and-polish and that what you have is great.

**Comment by lydia-duncan** [comment]
@mppf, @Kyle-B, and @thomasvandoren come immediately to mind at potential reviewers.

**Comment by lydia-duncan** [comment]
Note that I will make Brad's filerator rename, but I'm in the middle of reviewing Michael's I/O change so may not get to it for a little while.

File: modules/standard/FileSystem.chpl [hunk 1]
```diff
@@ -25,14 +25,60 @@
...
+   :proc:`isMount`
+   :proc:`sameFile`
+
+   .. _locale-state:
+
+   Locale State Functionality
+   --------------------------
+   :proc:`~locale.chdir`
+   :proc:`~locale.cwd`
+   :proc:`~locale.umask`
```
Comment (thomasvandoren) [inline modules/standard/FileSystem.chpl]: These links will show up as "chdir", "cwd", "umask". Would it make sense to leave off the tilde so they show up as "locale.chdir", etc. ?

Comment (bradcray) [inline modules/standard/FileSystem.chpl]: I think that's a good point.  I think we'll minimize confusion by using their fully-specified names here.

Comment (lydia-duncan) [inline modules/standard/FileSystem.chpl]: Okay, sounds good!

**Pull request merged**